src/tasks/task.ts

```diff
@@ -105,22 +105,21 @@ class Task implements TaskInterface {
   /**
    * ABSTRACT TEST METHOD
    */
-  private test(): Promise<TestResult> {
-    const mockOutput = { value: `test run ${this.taskData.id}` };
-    return Promise.resolve(mockOutput);
-  }
+  protected abstract test(): Promise<any>;
```

Comment:
Sadly not. The point of modelling all tasks with an abstract class, allows the `test()` to do whatever it wants, it might not return anything or it may return a wildly different shape to another task.